core/client/fs/src/main/java/alluxio/client/block/stream/LocalFileDataWriter.java

```diff
@@ -135,11 +135,11 @@ public void writeChunk(final ByteBuf buf) throws IOException {
 
   @Override
   public void cancel() throws IOException {
-    if (mStream.isClosed() || mStream.isCanceled()) {
-      return;
-    }
+    boolean cancelStream = !(mStream.isClosed() || mStream.isCanceled());
     try {
-      mStream.cancel();
+      if (cancelStream) {
+        mStream.cancel();
```

Comment:
I'm curious. In the cancel path, we call cancel on the stream, but we don't add to the closer, so the stream doesn't get closed. Is that intentional?

Reply:
it seems so. the documentation for the grpc blocking stream says `No other methods on this class can be called after this method has been called.`(for cancel). 

@bf8086  what do you think?

Reply:
Yeah once we cancel the gRPC request it is effectively closed.